File: src/main/com/ocaml/lang/OCamlLanguage.java
package com.ocaml.lang;

import com.intellij.lang.Language;
import org.jetbrains.annotations.NotNull;

/**
 * OCaml Language
 */
public final class OCamlLanguage extends Language {

    @NotNull
    public static final OCamlLanguage INSTANCE = new OCamlLanguage();

    private OCamlLanguage() {
        super("OCAML");
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "OCaml";
    }

    @Override
    public boolean isCaseSensitive() {
        return true;
    }
}

File: src/main/kotlin/com/ocaml/language/OCamlLanguage.kt
package com.ocaml.language

import com.intellij.lang.Language

/**
 * OCaml Language
 */
object OCamlLanguage : Language("OCaml") {
    private fun readResolve(): Any = OCamlLanguage
    override fun getDisplayName(): String = "OCaml"
    override fun isCaseSensitive(): Boolean = true
}